ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/Keystore.hs

```diff
@@ -34,15 +35,24 @@ import Pos.Core (AddressHash, addressHash)
 import Pos.Crypto (EncryptedSecretKey, PublicKey, encToPublic)
 
 import Ariadne.Wallet.Cardano.Kernel.DB.HdWallet (HdRootId(..))
-import Ariadne.Wallet.Cardano.Kernel.DB.InDb (InDb(..), fromDb)
+import Ariadne.Wallet.Cardano.Kernel.DB.InDb (InDb(..))
 import Ariadne.Wallet.Cardano.Kernel.Keystore.Persistence
   (getTempKeystorePath, peekInternalStorage, writeInternalStorage)
 import Ariadne.Wallet.Cardano.Kernel.Keystore.Types
-  (InternalStorage(..), isFilePath, isWallets)
-import Ariadne.Wallet.Cardano.Kernel.Keystore.Util (eskToPublicHash)
+  (InternalStorage(..), isWallets)
+import Ariadne.Wallet.Cardano.Kernel.Keystore.Util
+  (eskToPublicHash, walletIdToPublicHash)
 import Ariadne.Wallet.Cardano.Kernel.Types (WalletId(..))
 
-data Keystore = Keystore (MVar InternalStorage)
+data Storage
+  = Storage
+  { _sIstorage :: !InternalStorage
+  , _sFilePath :: !FilePath
```

Comment:
What's the purpose of putting `FilePath` into `MVar`? It's constant, why not make it a pure value?

Reply:
I didn't think through it deeply. I am going to change it such that I have this kind of constructor instead:

```
data Keystore
  = Keystore
  { _kstorage :: !(Mvar InternalStorage)
  , _sFilePath :: !FilePath
  }
```